File: examples/step02/tests/timer.pkg/reset.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2026 all rights reserved

# support
import sys
import time

# the package
from timer import Timer


def test() -> int:
    """Verify that reset() clears elapsed time and stops the timer."""
    t = Timer()
    t.start()
    time.sleep(0.01)
    t.stop()
    # reset clears both the accumulated time and the running flag
    t.reset()
    if t.running:
        print("FAILED: timer should not be running after reset()", file=sys.stderr)
        return 1
    if t.elapsed != 0.0:
        print("FAILED: elapsed time should be zero after reset()", file=sys.stderr)
        return 1
    # all done
    return 0


# main
if __name__ == "__main__":
    status = test()
    raise SystemExit(status)


# end of file

File: examples/step01/tests/timer.lib/reset.cc
// -*- C++ -*-
// -*- coding: utf-8 -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2026 all rights reserved

// support
#include <iostream>
// the library
#include <timer/Timer.h>


int
main()
{
    timer::Timer t;
    t.start();
    // burn some cycles so elapsed is non-zero before the reset
    volatile int dummy = 0;
    for (int i = 0; i < 1'000'000; ++i) {
        dummy += i;
    }
    t.stop();
    // reset clears the accumulated time and the running flag
    t.reset();
    if (t.running()) {
        std::cerr << "FAILED: timer should not be running after reset()" << std::endl;
        return 1;
    }
    if (t.elapsed() != 0.0) {
        std::cerr << "FAILED: elapsed time should be zero after reset()" << std::endl;
        return 1;
    }
    // all done
    return 0;
}


// end of file
